src/snowflake/connector/result_batch.py

```diff
@@ -258,6 +259,14 @@ def uncompressed_size(self) -> Optional[int]:
     def column_names(self) -> List[str]:
         return [col.name for col in self.schema]
 
+    @property
+    def use_sessions(self):
```

Comment:
On a second thought, I don't see the point of having this getter and setter, there's no error checking. Can't we just have `self.use_sessions`?

Reply:
Or what do you think about not even having `self.use_session`?
In `_download` we could just have it use the connection's session if connection is passed in.

Reply:
We'll need the flag to know when to close the response but I agree otherwise - I'll remove the getter and setter.

Reply:
I think we can get around the flag, here's the code that gets called: https://github.com/snowflakedb/snowflake-connector-python/blob/aaad5a4b4045d76330dcae9f90399d3ac4ab1ed3/src/snowflake/connector/vendored/requests/models.py#L951

Reply:
I took a closer look at the code that gets called. It does two things:
-  closes the "self.raw" file object, and
- release the connection
However, it turns out that when we read the file to the end, 
- the file object is closed
- the connection is released
(not in that order, but it still does everything that close does).

That being said, I think it would not hurt to explicitly call `close` here when we know we are done with the response (and we don't intend on accessing `response.raw` again). So I can remove the flag and have the `close` always run (however more often than not it will be a no-op)